### DISCUSSION TITLE
Keep both `=~`/`!~` and `like`/`not-like`?

### DISCUSSION THREAD

**Discussion by NotTheDr01ds** [nushell/nushell#14332] [Polls]
[#14702](https://github.com/nushell/nushell/pull/14072) introduced new operator-synonyms:

* `like` is the same as `=~`
* `not-like` is the same as `!~`

Opinions seem to be mixed in the comments on whether or not we should keep both forms.

**Comment by NotTheDr01ds** [comment]
I'm currently pretty firmly in the "keep both" camp at the moment, although that could change over time.  I *think* both have readability advantages in certain situations.  From `in`/`not-in`, I've seen that it's easy to view these as commands, rather than operators.  The sigils are much more clearly operators.

However, the `like` syntax is, as mentioned, more common to SQL users, and is a nice *option*.  It's also likely to be more semantically readable in some use-cases, but it's too early for me to know for sure.

**Reply by ghost** [reply]
i prefer symbols as well, but and will most likely ignore `like`'s existence. (my vote is for `keep symbols and idc about like`)

> the like syntax is, as mentioned, more common to SQL users

i think this might be more of a problem than a upside, since nu's `like` uses regex while sql's `like` uses [whatever this is](https://www.w3schools.com/SQL/sql_like.asp).  
while yes it is a better search-term i think it might result in people getting annoyed when a familiar command works the wrong way (i already get frusted from things which produce good errors like `if (foo == bar) return;` (java) vs `if foo == bar {return;}` (rust))

**Reply by weirdan** [reply]
>  it might result in people getting annoyed when a familiar command works the wrong way

I think you overestimate the annoyance. Nushell doesn't support `select *` nor `select ... from`, the two most common parts of an SQL query, and yet people seem to be ok with that.

**Comment by weirdan** [comment]
Unfortunately, no vote options are proposed for 'keep A, don't care about B'. For me, `=~` is ok readability-wise but is annoying to type as it requires `Shift` and I can't seem to remember whether the tilde comes before or after the equals sign. So my answer would be 'Keep `like`; don't care about sigils'.

**Reply by Taywee** [reply]
Agreed.  I'm in the inverse camp of "keep `=~`, I don't care about `like`".

It would be nice if GitHub polls supported multiple choices.

**Comment by Taywee** [comment]
I find `like` and `not-like` an odd choice of words.  I can't think of a single other place where "like" means "regex match", but at least 3 other programming languages I use regularly where `=~` means "regex match".

I don't care about `like`, but I'm firmly against removing `=~`.  Most users of nushell are people who care about structured data, who will be most likely programmers or at least users of some scripting languages.

**Reply by hustcer** [reply]
You have my vote

**Reply by fdncred** [reply]
I added `like` because it's closer to sql, while not regex it's close due to using `%` as wildcards and other things. Then I needed to add the inverse so `not-like` seemed reasonable. I find `=~` and `!~` a pain to type.

**Reply by Taywee** [reply]
That does make sense, I just prefer the consistency.  Personally, I don't like `=~` either; I prefer a regex type with dedicated regex functions/methods (eg. `my_regex.match(string)`), but that's not really much of an option here.

**Comment by Or1g3n** [comment]
100% in the camp of both because to me one is a shorthand using symbols while the other is verbose and easy to understand even if you have never written a programming language. In my opinion ```like``` and ```not-like``` is a simple way to lower the barrier to entry and avoids use of tilde which at least on my keyboard will cause carpal tunnel over time :). Also, since data engineers will be drawn to nushell given it's structured data ethos, like and not-like are super common in SQL land.
